File: android/src/main/kotlin/net/lagerstroemia/naver_login_sdk/api/NaverLoginSdkConstant.kt
package net.lagerstroemia.naver_login_sdk

object NaverLoginSdkConstant {
    const val channelNameMethod: String = "naver_login_sdk_method"
    const val channelNameEvent: String = "naver_login_sdk_event"

    object Key {
        const val initialize: String = "initialize"
        const val authenticate: String = "authenticate"

        object OAuthLoginCallback {
            const val onSuccess: String = "onSuccess"
            const val onFailure: String = "onFailure"
            const val onError: String = "onError"
        }
    }

    object Value {
        const val clientId: String = "clientId"
        const val clientSecret: String = "clientSecret"
        const val clientName: String = "clientName"

        object OAuthLoginCallback {
            const val errorCode: String = "errorCode"
            const val message: String = "message"
        }
    }
}

File: ios/Classes/Api/NaverLoginSdkConstant.swift
//
//  NaverLoginSdkConstant.swift
//  naver_login_sdk
//
//  Created by Lagerstroemia on 11/17/24.
//

import Foundation

struct NaverLoginSdkConstant {
    static let channelNameMethod: String = "naver_login_sdk_method"
    static let channelNameEvent: String = "naver_login_sdk_event"
    
    struct Key {
        static let initialize: String  = "initialize"
        static let authenticate: String = "authenticate"
        static let logout: String = "logout"
        static let release: String = "release"
        static let profile: String = "profile"
        
        struct NaverLoginEventCallback {
            static let onSuccess: String = "onSuccess"
            static let onFailure: String = "onFailure"
            static let onError: String = "onError"
        }
    }
    
    struct Value {
        static let urlScheme: String = "urlScheme"
        static let clientId: String = "clientId"
        static let clientSecret: String = "clientSecret"
        static let clientName: String = "clientName"
    }
}
